src/ConductorSharp.Engine/Util/ExpressionUtil.cs

```diff
@@ -29,7 +29,59 @@ public static Type ParseToType(Expression expression)
             return ((PropertyInfo)taskSelectExpression.Member).PropertyType;
         }
 
-        public static JObject ParseToParameters(Expression expression)
+        public static JObject ParseToParameters(Expression expression) => ParseObjectInitialization(expression);
+
+        private static object ParseExpression(Expression expression)
+        {
+            if (expression is ConstantExpression cex)
+                return Convert.ToString(cex.Value);
+
+            if (expression is UnaryExpression uex && uex.Operand is ConstantExpression ccex)
+            {
+                var converted = Convert.ToString(ccex.Value);
+
+                //if (ccex.Value is bool)
+                //    return converted.ToLowerInvariant();
+
+                return ccex.Value;
+            }
+
+            // Handle boxing
+            if (expression is UnaryExpression unaryEx && unaryEx.NodeType == ExpressionType.Convert)
+                return CreateExpressionString(unaryEx.Operand);
+
+            if (
+                expression is MethodCallExpression methodExpression
+                && methodExpression.Method.Name == nameof(string.Format)
+                && methodExpression.Method.DeclaringType == typeof(string)
+            )
+            {
+                var expressionStrings = methodExpression.Arguments.Skip(1).Select(CompileMemberOrNameExpressions).ToArray();
+                var formatExpr = methodExpression.Arguments[0] as ConstantExpression;
+                if (formatExpr == null)
+                    throw new Exception("string.Format with non constant format string is not supported");
+                var formatString = (string)formatExpr.Value;
+                return string.Format(formatString, expressionStrings);
+            }
+
+            if (expression is NewExpression || expression is MemberInitExpression)
+                return ParseObjectInitialization(expression);
+
+            if (expression is NewArrayExpression newArrayExpression)
+                return ParseArrayInitalization(newArrayExpression);
+
+            return CompileMemberOrNameExpressions(expression);
+        }
+
+        private static JArray ParseArrayInitalization(NewArrayExpression newArrayExpression)
+        {
+            if (newArrayExpression.NodeType != ExpressionType.NewArrayInit)
+                throw new Exception("Only dimensionless array initialization is supported");
+
+            return new JArray(newArrayExpression.Expressions.Select(ParseExpression));
```

Comment:
This seems to return [ 1,2,3] as ["1", "2", "3"]. The reason is 

```
 if (expression is ConstantExpression cex)
                return Convert.ToString(cex.Value);
```
@boma96 
We can extract the correct type, but this might break something somewhere. However, most frameworks seem to be equipped to handle string to int conversion automatically if it is really convertible. I think we should do this, what do you think?

Reply:
It will also break our tests

Reply:
Thank you for pointing this out, did not notice it

Reply:
Ok, will make separate PR for this